### DISCUSSION TITLE
Hashicorp vault

### DISCUSSION THREAD

**Discussion by zachhutchins** [mia-platform/community#402] [Q&A] [answered]
### What product are you having troubles with?

Console

### What Console version are you using?

1.95.2-hotfix.0

### Description

Mia-Platform v12.2.0

Added provider for using Vault to manage environment secrets as documented here https://docs.mia-platform.eu/docs/console/project-configuration/manage-environment-variables/manage-environment-variables-with-vault

Setup vault as specified in the documentation
Added provider using token auth
Added provider to cms

### Actual Outcome

After adding provider console is throwing error 'selected storage is not valid'

This feels like that vault is not capable of being used, I would expect an error stating authentication/authorization error or configuration error if it was one of those.

![image](https://github.com/mia-platform/community/assets/136849183/29fda3e1-88cd-44e5-b268-0d7f4fc0108b)

### Expected Outcome

To use vault to add environment variables to.

**Comment by galatifederico** [comment]
Hi @zachhutchins can you share the response received by the _/api/projects/{projectId}/variables_ endpoint?

**Comment by mattia-fumo** [comment]
As stated [here](https://docs.mia-platform.eu/docs/console/project-configuration/manage-environment-variables/manage-environment-variables-with-vault#use-vault-variables-inside-a-project):

> Projects that use Vault as the secret provider cannot manage their variables using the Console interface but they need to be manually edited from the Vault instance

The Console, at the moment, does not support the editing and visualization of Vault secrets, that's why you are receiving this error. If you can save the configuration of the Project correctly then the Vault provider is correctly configured and working. To manage the secrets please use the Vault UI.

**Comment by zachhutchins** [comment]
@mattia-fumo Thanks for getting back to me! Appreciate your help on this.

That definitely clears things up a bit, saving the configuration is possible on our end. One detail I did not think was of importance, but I am thinking so now, we are using a pull-based Deployment Option. The desire was to have interpolation happen from vault so we could get those values via the orchestration plugin. I am guessing that is not an option here?

**Reply by mattia-fumo** [reply, marked as answer]
No, indeed. At the moment our services are not fetching any secret value (only keys) from Vault. Actual values are available in the cluster where workloads can mount the generated Kubernetes secret to read the values.

**Reply by zachhutchins** [reply]
Just for clarification, if we are doing a pull-based deployment then then those values would not be available in the cluster unless we put them there ourselves? You are referencing value availability in a push-based deployment model?

**Reply by ericlanderson** [reply]
In previous versions of the Mia Platform docs there were references to installing and using the `external-secrets` operator to sync secrets from Vault to Kubernetes native Secrets. When you reference 
> mount the generated Kubernetes secret to read the values.

Is that still the expectation?

**Reply by mattia-fumo** [reply]
> Just for clarification, if we are doing a pull-based deployment then then those values would not be available in the cluster unless we put them there ourselves? You are referencing value availability in a push-based deployment model?

Push or pull make no difference, since the same Kubernetes resources are generated

**Reply by mattia-fumo** [reply]
@ericlanderson, nothing changed from previous versions, at least architecturally. So yes.
